src/game/puppyprint.c

```diff
@@ -1226,6 +1228,62 @@ void get_char_from_byte(u8 letter, s32 *textX, s32 *textY, u8 *spaceX, s8 *offse
         case 'y': if (font == FONT_DEFAULT) *offsetY = 1; break;
     }
 }
+// This is where the deferred printing will be stored. When text is made, it will store text with an 12 byte header, then the rest will be the text data itself.
+// The first 4 bytes of the header will be the X and Y pos
+// The next 4 bytes will be the current envcolour set by print_set_envcolour
+// Then the string length, text alignment, amount and font each get a byte.
+// The data afterwards is the text data itself, using the string length byte to know when to stop.
+void print_small_text_buffered(s32 x, s32 y, const char *str, u8 align, s32 amount, u8 font) {
+    u8 strLen = MIN((signed)strlen(str), 255);
+    // Compare the cursor position and the string length, plus 11 (header size) and return if it overflows.
+    if (sPuppyprintTextBufferPos + strLen + 11 > sizeof(sPuppyprintTextBuffer))
+        return;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos] = (x >> 8) & 0xFF;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 1] = x & 0xFF;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 2] = (y >> 8) & 0xFF;
+    sPuppyprintTextBuffer[sPuppyprintTextBufferPos + 3] = y & 0xFF;
```

Comment:
I don't expect these shifts to work properly on negative inputs. Maybe try casting the starting offset of sPuppyptintTextBuffer to a struct pointer and treat the header like a struct from there.